### ISSUE TITLE
Looking to add stream game name to filename

### ISSUE THREAD

**Issue by Nach0-Stallion** [lauwarm/docker-streamlink-recorder#20]
Hello and happy new year!
I am trying to get the output filename to include the game name as part of the file, i have tried the category & game variables but either get errors or the file just has 'game' in the filename
the variables in the script file use $variables and i have tried adding them, but im clueless with this im afraid. and not sure if i should be using "{category}" or $category somewhere.
id just like to be able to have "Streamername - Gametheyareplaying - Date .mp4"
Really appreciate any assistance you can offer me!

**Comment by lauwarm** [comment]
Hello @Nach0-Stallion and a happy new year to you, too!

Are you trying to set the "Category & Game" by yourself? Or is _streamlink_ itself reporting this?

If it is the former, you could do a dirty hack and add the game title to the _streamName_ until I worked out a way to maybe add generic tags.

In case we are talking about the latter one, I probably need to rewrite the script. Currently the only dynamic part is the _Date_. Everything else is static and defined by environment variables.

**Comment by Nach0-Stallion** [comment]
Hi!, Thanks for getting back to me
I was looking into a way to add the game name to the files as my folder is flooded with "streamername-date.mkv" so trying to pick out what video for what game the streamer is playing means i have to launch each one, not a huge deal but i did see from streamlink that its capable of reading the stream title and category (which is used for the game name apparently)
i looked in the manual and in the metadata section of [https://streamlink.github.io/cli.html](https://streamlink.github.io/cli.html) it lists :

> <html><body>
<div class="table-wrapper">

category | The stream's category, eg. the name of a game being played, a music genre, etc.
-- | --
game | Alias for category.
url | The resolved URL of the stream.
time | The current timestamp. Can optionally be formatted via {time:format}. The format parameter string is passed to Python's datetime.strftime() method, so all the usual time directives are available. The default format is %Y-%m-%d_%H-%M-%S.

</div>
<p>Examples:</p>
<div class="highlight-console notranslate"><div class="highlight"><pre><span></span><span class="gp">$ </span>streamlink --title <span class="s2">"{author} - {category} - {title}"</span> &lt;URL&gt; <span class="o">[</span>STREAM<span class="o">]</span>
<span class="gp">$ </span>streamlink --output <span class="s2">"~/recordings/{author}/{category}/{id}-{time:%Y%m%d%H%M%S}.ts"</span> &lt;URL&gt; <span class="o">[</span>STREAM<span class="o">]</span>
</pre></div>
</div>
</body>
</html>

So i thought that means it can pull the game title (if listed from streamer) to then be able to have "streamername-gamename-date.mkv
so if streamlink itself can do it, adding it to this i thought was possible, but im so noob at this stuff.

Thank you again!

**Comment by lauwarm** [comment]
Hey, 

thank you for the Information. I had no idea about all those metadata which is available from _streamlink_.
I will have a look at this and try to update the script so it contains those _categories_.

It's just going to take some time.

**Comment by Nach0-Stallion** [comment]
Hi!
I didn't realise it would need to be implemented , I thought I was just not using it correctly 😬
If it's something you can, that would be amazing, but please no rush at all!
Thank you

**Comment by lauwarm** [comment]
Hey @Nach0-Stallion,

I got around to update the Docker Image. The Branch is called _future-development_.
I tried it with two twitch streams and it seems to be working. There could be some problems though:

- Extended ASCII characteres could mean that filenames are not readable
- Once the stream recording started, the _category_ is set. Meaning if a streamer switches the category nothing will happen to the filename

If you encouter bugs just let me know. 

To run the _future-development_ Branch you would use this command
```
docker run -v /path/to/download-folder:/home/download -e streamLink='twitch.tv/twitch' -e streamQuality='best' -e streamOptions='--twitch-disable-hosting' lauwarm/streamlink-recorder:future-development
```

Till then, have a nice weekend
